src/main/java/medic/gateway/alert/WebappPoller.java

```diff
@@ -61,6 +61,13 @@ public SimpleResponse pollWebapp() throws JSONException, MalformedURLException {
 		return response;
 	}
 
+	public Boolean moreMessagesToSend() {
+		// REVIEWER: we could change db calls to get MAX+1, and use that
+		// to determine if there are *truly* more... thoughts?
```

Comment:
The only downside is occasionally sending a null request, right? I think I'm ok with it as is, because there's no simple solution.

The change you suggest is fine though it always leads to slightly complicated code when you have to drop the last one.

Another option is to change the `pollWebapp` code so it doesn't send the request if the counts are 0. Unfortunately this isn't very clean either because we still want to poll at least once to see if there's anything to receive, so you'd also need to make a change to the `WakefulService` to pass in whether or not this is the first time through the polling loop.

I think I'd choose to leave it as is but will leave the final decision with you.